contracts/MechMarketplace.sol

```diff
@@ -194,21 +169,26 @@ contract MechMarketplace is IErrorsMarketplace {
             revert ZeroValue();
         }
 
+        // TODO What to do if there is rateDIff leftovers? Keep as a fee or record into sender's map?
+        uint256 rateDiff;
+        if (actualDeliveryRate > deliveryRate) {
+            rateDiff = actualDeliveryRate - deliveryRate;
+        }
```

Comment:
in case of fixed price it just gets consumed, otherwise all we care for is lower or equal than max (because it won't be spent anyway)